### PR TITLE
llvm 16.0.0rc1

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Plus, while we're here, some small improvements.

### CODE REVIEW

File: Formula/llvm.rb [hunk 1]
```diff
@@ -2,8 +2,9 @@ class Llvm < Formula
   desc "Next-gen compiler infrastructure"
   homepage "https://llvm.org/"
   # NOTE: `ccls` will need rebuilding on every version bump.
-  url "https://github.com/llvm/llvm-project/releases/download/llvmorg-15.0.7/llvm-project-15.0.7.src.tar.xz"
-  sha256 "8b5fcb24b4128cf04df1b0b9410ce8b1a729cb3c544e6da885d234280dedeac6"
+  url "https://github.com/llvm/llvm-project/releases/download/llvmorg-16.0.0-rc1/llvm-project-16.0.0rc1.src.tar.xz"
+  version "16.0.0rc1"
```
Comment (carlocab) [inline Formula/llvm.rb:6]: Our version parser is picking this up at `6.0.0rc1`, which is wrong.

The errant parser seems to be [this one](https://github.com/Homebrew/brew/blob/7a5f6143d3e7447a64255c548d2e51138c83acd3/Library/Homebrew/version.rb#L466-L470).
```ruby
    # e.g. https://registry.npmjs.org/@angular/cli/-/cli-1.3.0-beta.1.tgz
    # e.g. https://github.com/dlang/dmd/archive/v2.074.0-beta1.tar.gz
    # e.g. https://github.com/dlang/dmd/archive/v2.074.0-rc1.tar.gz
    # e.g. https://github.com/premake/premake-core/releases/download/v5.0.0-alpha10/premake-5.0.0-alpha10-src.zip
    StemParser.new(/[.-vV]?(#{NUMERIC_WITH_DOTS}#{PRERELEASE_SUFFIX})/),
```

Comment (carlocab) [reply Formula/llvm.rb:6]: Homebrew/brew#14471

**Comment by carlocab** [comment]
To-do: see if we can ship a configuration file instead of setting `-DDEFAULT_SYSROOT`. That'll allow Xcode-only systems to install bottles too.

https://clang.llvm.org/docs/UsersManual.html#configuration-files

**Comment by carlocab** [comment]
> To-do: see if we can ship a configuration file instead of setting `-DDEFAULT_SYSROOT`. That'll allow Xcode-only systems to install bottles too.
> 
> https://clang.llvm.org/docs/UsersManual.html#configuration-files

We can set `CLANG_CONFIG_FILE_SYSTEM_DIR`, but this will break relocatability of the bottles because this will hardcode some path inside `HOMEBREW_PREFIX`. I'm planning to upstream a patch that will allow us to use a relative path, but this will probably take me a while.

**Comment by ZhongRuoyu** [comment]
16.0.0 is [out](https://github.com/llvm/llvm-project/releases/tag/llvmorg-16.0.0). Time to resurrect this, maybe?

**Comment by carlocab** [comment]
Pushed the important changes to #126023.

**Pull request closed without merging**